needy/needy_configuration.py

```diff
@@ -0,0 +1,53 @@
+import os
+import json
+import logging
+
+from .caches.directory import Directory
+
+
+class NeedyConfiguration:
+    def __init__(self, base_path, allow_user_config=True):
+        self.__base_path = base_path
+        self.__allow_user_config = allow_user_config
+
+        self.__configuration = self._evaluate_needyconfigs(self._get_needyconfigs(base_path, self.__allow_user_config))
+
+    @staticmethod
+    def _get_needyconfigs(base_path, allow_user_config):
+        ret = []
+        path = base_path
+
+        candidates = []
+
+        while path and path != '/':
+            needyconfig = os.path.join(path, '.needyconfig')
+            candidates.append(needyconfig)
+            path = os.path.dirname(path)
+
+        user_config = os.path.join(os.path.expanduser('~'), '.needyconfig')
+        if allow_user_config and user_config not in candidates:
+            candidates.append(user_config)
+
+        for c in candidates:
+            if os.path.exists(c):
+                logging.debug('using .needyconfig at {}'.format(c))
+                ret.append(c)
+
+        return ret
+
+    @staticmethod
+    def _evaluate_needyconfigs(paths):
+        config = dict()
+        for c in reversed(paths):
+            with open(c, 'r') as f:
+                config.update(json.loads(f.read()))
```

Comment:
Do you really want to just `update` the dict? I think for caches, a union of the list makes a lot more sense. So maybe evaluate lazily and expose a method that allows a policy to be specified: `config.union('build-caches')` or `config.first('someotherthing')`